internal/providers/github/properties/fetcher.go

```diff
@@ -0,0 +1,97 @@
+//
+// Copyright 2024 Stacklok, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package properties
+
+import (
+	"context"
+	"fmt"
+	go_github "github.com/google/go-github/v63/github"
+	"github.com/stacklok/minder/internal/entities/properties"
+	minderv1 "github.com/stacklok/minder/pkg/api/protobuf/go/minder/v1"
+)
+
+type PropertyWrapper func(ctx context.Context, ghCli *go_github.Client, name string) (map[string]any, error)
+
+type PropertyFetcher interface {
+	WrapperForProperty(propertyKey string) PropertyWrapper
+	AllPropertyWrappers() []PropertyWrapper
+	OperationalProperties() []string
+	GetName(props *properties.Properties) (string, error)
+}
+
+type PropertyFetcherFactory interface {
+	EntityPropertyFetcher(entType minderv1.Entity) PropertyFetcher
+}
```

Comment:
These interfaces are not supposed to be used outside github, but I found it easier to decouple fetching of the properties and the operational attributes into objects that sit in the provider rather than having separate hardcoded lists. It will make handling of more properties easier and makes the provider code oblivious to what properties are operational and how to fetch properties. All that would be needed to add a new entity is to extend the PropertyFetcherFactory for new entities and implement the PropertyFetcher for a new entity.

Reply:
If it's not meant to be used outside of GitHub, can you add that as a comment and/or prepend the word `GitHub` in front?

Reply:
I added a prefix as to not stutter